core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -200,9 +200,10 @@ public ExpireSnapshots expireSnapshots() {
 
   @Override
   public void commitTransaction() {
-    Preconditions.checkState(lastBase != current,
-        "Cannot commit transaction: last operation has not committed");
-
+    if (lastBase == current) {
+      LOG.info("Nothing to commit.");
+      return;
+    }
```

Comment:
This isn't correct:
* Skipping the commit should be done in `TableOperations`, not here. This logic is incorrect because `lastBase` is not the current table state, it is the state after the last operation in the transaction. This logic would skip the rest of the operations in a transaction if the last one didn't make any changes.
* This still needs to check that the last operation had `commit` called on it. Otherwise, incomplete operations could be added and would be ignored instead of raising exceptions.

Reply:
I agree with this, it changes the previous commit behaviour.